File: iis-vsts/develop/developing-for-ftp/how-to-use-managed-code-visual-basic-to-create-a-simple-ftp-logging-provider/samples/sample2.vb
Imports System
Imports System.IO
Imports Microsoft.Web.FtpServer

Namespace FtpLogging
    Public Class FtpLogDemo
        Inherits BaseProvider
        Implements IFtpLogProvider
        Public Sub Log(ByVal loggingParameters As FtpLogEntry) _
            Implements IFtpLogProvider.Log
            ' Note: You would add your own custom logic here.
            ' Open the log file for output.
            Dim sw As StreamWriter = New StreamWriter( _
                "C:\logfiles\myftpsite\myftplog.log", True)
            ' Retrieve the current date and time for the log entry.
            Dim dt As DateTime = DateTime.Now
            ' Retrieve the user name.
            Dim un As String = loggingParameters.UserName
            ' Write the log entry to the log file.
            sw.WriteLine( _
                String.Format("{0}" & vbTab & "{1}" & vbTab & "{2}" & _
                vbTab & "{3}" & vbTab & "{4}" & vbTab & "{5}", _
                dt.ToShortDateString(), _
                dt.ToLongTimeString(), _
                loggingParameters.RemoteIPAddress, _
                IIf((un.Length = 0), "-", un), _
                loggingParameters.Command, _
                loggingParameters.SessionId))
            sw.Dispose()
        End Sub
    End Class
End Namespace

File: iis-vsts/develop/developing-for-ftp/how-to-use-managed-code-c-to-create-a-simple-ftp-logging-provider/samples/sample2.cs
using System;
using System.IO;
using Microsoft.Web.FtpServer;

namespace FtpLogging
{
    public class FtpLogDemo : BaseProvider,
        IFtpLogProvider
    {
        void IFtpLogProvider.Log(FtpLogEntry loggingParameters)
        {
            // Note: You would add your own custom logic here.
            // Open the log file for output.
            using (StreamWriter sw =
                new StreamWriter(@"C:\logfiles\myftpsite\myftplog.log",true))
                {
                // Retrieve the current date and time for the log entry.
                DateTime dt = DateTime.Now;
                // Retrieve the user name.
                string un = loggingParameters.UserName;
                // Write the log entry to the log file.
                sw.WriteLine("{0}\t{1}\t{2}\t{3}\t{4}\t{5}",
                    dt.ToShortDateString(),
                    dt.ToLongTimeString(),
                    loggingParameters.RemoteIPAddress,
                    (un.Length == 0) ? "-" : un,
                    loggingParameters.Command,
                    loggingParameters.SessionId);
            }
        }
    }
}